### PR TITLE
Add SPM Support and new demo build target with SPM dependencies

### PR DESCRIPTION
**Pull request by dingyishen-amazon**
## ℹ️ Description
* Add SPM instructions on README
* Add new demo build target with SPM dependencies

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
    - [ ] README update
    - [ ] CHANGELOG update
    - [ ] guides update
- [ ] This change requires a dependency update
    - [ ] Amazon Chime SDK Media
    - [ ] Other (update corresponding legal documents)

## 🧪 How Has This Been Tested?
Tested with Jenkins build job: http://ec2-54-152-250-229.compute-1.amazonaws.com:4080/view/%F0%9F%92%BC%20iOS%20SDK%20SPM/job/Pre-release%20-%20Build%20demo%20AmazonChimeSDKDemoSPM%20target/
Build succeeded.

### Additional Manual Test
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Rotate screen back and forth

## 📱 Screenshots, if available
*provide screenshots/video record if there's a UI change in demo app*

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dylonChime — APPROVED** [review]

**Review by hokyungh — COMMENTED** [review]
Were you able to test the build worked with screen share and everything else?

File: README.md [hunk 1]
```diff
@@ -102,6 +102,19 @@ For the purpose of setup, your project's root folder (where you can find your `.
 <img src="./media/cocoapods_machine_learning.png" alt="image" width="80%"/>
 </p>
 
+### From SPM
```
Comment (georgezy-amzn) [inline README.md:105]: Also need to add SPM to line 43.

Comment (dingyishen-amazon) [reply README.md:105]: Good catch.

**Comment by dingyishen-amazon** [comment]
> Were you able to test the build worked with screen share and everything else?

Yes.

**Review by georgezy-amzn — APPROVED** [review]

**Review by hokyungh — APPROVED** [review]

**Pull request merged**